src/transformers/models/ernie_m/modeling_ernie_m.py

```diff
@@ -0,0 +1,1042 @@
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+    @add_start_docstrings_to_model_forward(ERNIE_M_INPUTS_DOCSTRING.format("batch_size, sequence_length"))
+    @add_code_sample_docstrings(
+        processor_class=_TOKENIZER_FOR_DOC,
+        checkpoint=_CHECKPOINT_FOR_DOC,
+        output_type=SequenceClassifierOutput,
+        config_class=_CONFIG_FOR_DOC,
+    )
+    def forward(
+        self,
+        input_ids=None,
+        attention_mask=None,
+        position_ids=None,
+        head_mask=None,
+        inputs_embeds=None,
+        past_key_values=None,
+        use_cache=None,
+        output_hidden_states=None,
+        output_attentions=None,
+        return_dict=True,
+        labels=None,
+    ):
+        r"""
+        labels (`torch.LongTensor` of shape `(batch_size,)`, *optional*):
+            Labels for computing the sequence classification/regression loss. Indices should be in `[0, ...,
+            config.num_labels - 1]`. If `config.num_labels == 1` a regression loss is computed (Mean-Square loss), If
+            `config.num_labels > 1` a classification loss is computed (Cross-Entropy).
+        """
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        outputs = self.ernie_m(
+            input_ids,
+            attention_mask=attention_mask,
+            position_ids=position_ids,
+            head_mask=head_mask,
+            inputs_embeds=inputs_embeds,
+            past_key_values=past_key_values,
+            output_hidden_states=output_hidden_states,
+            output_attentions=output_attentions,
+            return_dict=return_dict,
+        )
+
+        pooled_output = outputs[1]
+
+        pooled_output = self.dropout(pooled_output)
+        logits = self.classifier(pooled_output)
+
+        loss = None
+        if labels is not None:
+            if self.config.problem_type is None:
+                if self.num_labels == 1:
+                    self.config.problem_type = "regression"
+                elif self.num_labels > 1 and (labels.dtype == torch.long or labels.dtype == torch.int):
+                    self.config.problem_type = "single_label_classification"
+                else:
+                    self.config.problem_type = "multi_label_classification"
+
+            if self.config.problem_type == "regression":
+                loss_fct = MSELoss()
+                if self.num_labels == 1:
+                    loss = loss_fct(logits.squeeze(), labels.squeeze())
+                else:
+                    loss = loss_fct(logits, labels)
+            elif self.config.problem_type == "single_label_classification":
+                loss_fct = CrossEntropyLoss()
+                loss = loss_fct(logits.view(-1, self.num_labels), labels.view(-1))
+            elif self.config.problem_type == "multi_label_classification":
+                loss_fct = BCEWithLogitsLoss()
+                loss = loss_fct(logits, labels)
```

Comment:
It is very nice that the model supports different training strategies, can you maybe add small tests, testing each training scenarios (by running a dummy inference + `loss.backward` to make sure the gradient computation works fine)

Reply:
Do you meant to add multiple tests(eg. Regression, single_label_clf) etc?

Reply:
yes, this is for example what we did in BLIP: https://github.com/huggingface/transformers/blob/833174c9290e70e6dc7f0d648dc0e098bf87b0b6/tests/models/blip/test_modeling_blip.py#L646

Reply:
I believe there is already a test like BLIP which tests all model downstream heads(not like different tests for different loss functions) I remember it was always failing. Could you please check again if that's the case or not since I am pretty sure there was a test.

Reply:
Here are the results of pytest, 
```
tests/models/ernie_m/test_modeling_ernie_m.py::ErnieMModelTest::test_training PASSED                                        [ 96%]
tests/models/ernie_m/test_modeling_ernie_m.py::ErnieMModelTest::test_training_gradient_checkpointing PASSED                 [ 98%]
```
As you can see it's already testing both `test_training` and `test_training_gradient_checkpointing` .

Reply:
I see, those are some defaults tests we run in the CI runners, but is it testing all the possible `config.problem_type` values?

Reply:
In any case I think this is probably fine, as this is done also for ernie & I see that there is a test for trsting the forward pass of this class